atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/AbstractDefinition.java

```diff
@@ -52,6 +71,65 @@ public void sweepStrategy(SweepStrategy strategy) {
         this.sweepStrategy = strategy;
     }
 
+    public void ignoreHotspottingChecks() {
+        ignoreHotspottingChecks = true;
+    }
+
+    public boolean shouldIgnoreHotspottingChecks() {
+        return ignoreHotspottingChecks;
+    }
+
+    public void rangeScanAllowed() {
+        rangeScanAllowed = true;
+    }
+
+    public boolean isRangeScanAllowed() {
+        return rangeScanAllowed;
+    }
+
+    public boolean isExplicitCompressionRequested(){
+        return explicitCompressionRequested;
+    }
+
+    public void explicitCompressionRequested() {
+        explicitCompressionRequested = true;
+    }
+
+    public int getExplicitCompressionBlockSizeKB() {
+        return explicitCompressionBlockSizeKB;
+    }
+
+    public void explicitCompressionBlockSizeKB(int blockSizeKB) {
+        explicitCompressionBlockSizeKB = blockSizeKB;
+    }
+
+    public void negativeLookups() {
+        negativeLookups = true;
+    }
+
+    public boolean hasNegativeLookups() {
+        return negativeLookups;
+    }
+
+    public void appendHeavyAndReadLight() {
+        appendHeavyAndReadLight = true;
+    }
+
+    public boolean isAppendHeavyAndReadLight() {
+        return appendHeavyAndReadLight;
+    }
+
     protected abstract ConflictHandler defaultConflictHandler();
+
+    protected static boolean canHotspotIfFirstRowComp(NameComponentDescription firstRowComp) {
+        if (CRITICAL_ROW_TYPES.contains(firstRowComp.getType())) {
```

Comment:
This should only be called from validate methods (and was added here to share between the two validates).  I can rename it to `validateFirstRowComponent` to make it clearer.